Test Fails if JUnitPerfTestRequirement(executionsPerSec) is not explicitly assigned

Hello,

Whilst testing this tool I encountered the following error consistently:

```
java.lang.AssertionError: Error threshold not achieved
Expected: is <true>
     but: was <false>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at com.github.noconnor.junitperf.statements.PerformanceEvaluationStatement.assertThresholdsMet(PerformanceEvaluationStatement.java:69)
	at com.github.noconnor.junitperf.statements.PerformanceEvaluationStatement.evaluate(PerformanceEvaluationStatement.java:65)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
```

I notice that by default in *JUnitPerfTestRequirement*
```  
// Expected test throughput (executions per second)
  int executionsPerSec() default 1;
```

But my test always fails unless I add the following annotation (documented as optional):

```
@JUnitPerfTestRequirement(executionsPerSec = 99)
```

Here is my simple test, with the JUnitPerfTestRequirement annotation,  without it fails.

```java
package com.test;

import static org.junit.Assert.assertTrue;

import java.io.IOException;
import java.net.InetSocketAddress;
import java.net.Socket;

import org.junit.Rule;
import org.junit.Test;

import com.github.noconnor.junitperf.JUnitPerfRule;
import com.github.noconnor.junitperf.JUnitPerfTest;
import com.github.noconnor.junitperf.JUnitPerfTestRequirement;

public class TestSocketConnect {

	@Rule
	public JUnitPerfRule perfTestRule = new JUnitPerfRule();

	@Test
	@JUnitPerfTest(threads = 10, durationMs = 10_000, warmUpMs = 1_000, maxExecutionsPerSecond = 100)
	@JUnitPerfTestRequirement(executionsPerSec = 99)
	public void test2() {
		Socket socket = new Socket();
		try {
			socket.connect(new InetSocketAddress("www.google.com", 80), 1000);
			assertTrue(socket.isConnected());
		} catch (IOException e) {
		} finally {
			try {
				socket.close();
			} catch (IOException e) {
				e.printStackTrace();
			}
		}
	}
}
```

--- a/README.md
+++ b/README.md
@@ -248,9 +248,9 @@ public JUnitPerfRule perfTestRule = new JUnitPerfRule(new CustomStatisticsCalcul
 
 ## Build Instructions
 
-To compile this project and run tests execute the following command from the root project directory: `gradle clean test`
+To compile this project and run tests execute the following command from the root project directory: `./gradlew clean test`
 
-To generate a library jar execute: `gradle clean assemble`
+To generate a library jar execute: `./gradlew clean assemble -x sign` or `./gradlew clean jar` 
 
 **Intellij 14 Setup**
 

--- a/src/main/java/com/github/noconnor/junitperf/data/EvaluationContext.java
+++ b/src/main/java/com/github/noconnor/junitperf/data/EvaluationContext.java
@@ -16,6 +16,7 @@ import com.google.common.primitives.Ints;
 import static com.google.common.base.Preconditions.checkNotNull;
 import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Maps.newTreeMap;
+import static java.util.Collections.emptyMap;
 import static java.util.Objects.nonNull;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
@@ -93,6 +94,9 @@ public class EvaluationContext {
       isSuccessful = isThroughputAchieved && isErrorThresholdAchieved && noLatencyPercentileFailures();
     } else {
       isSuccessful = true;
+      isThroughputAchieved = true;
+      isErrorThresholdAchieved = true;
+      percentileResults = emptyMap();
     }
   }
 

--- a/src/test/examples/com/github/noconnor/junitperf/examples/ExampleSuccessTests.java
+++ b/src/test/examples/com/github/noconnor/junitperf/examples/ExampleSuccessTests.java
@@ -1,5 +1,35 @@
 package com.github.noconnor.junitperf.examples;
 
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import org.junit.Rule;
+import org.junit.Test;
+import com.github.noconnor.junitperf.JUnitPerfRule;
+import com.github.noconnor.junitperf.JUnitPerfTest;
+
+import static org.junit.Assert.assertTrue;
+
 public class ExampleSuccessTests {
 
+  @Rule
+  public JUnitPerfRule perfRule = new JUnitPerfRule();
+
+  @Test
+  @JUnitPerfTest(threads = 10, durationMs = 10_000, warmUpMs = 1_000, maxExecutionsPerSecond = 100)
+  public void whenNoRequirementsArePresent_thenTestShouldAlwaysPass() {
+    Socket socket = new Socket();
+    try {
+      socket.connect(new InetSocketAddress("www.google.com", 80), 1000);
+      assertTrue(socket.isConnected());
+    } catch (IOException e) {
+      // IGNORE
+    } finally {
+      try {
+        socket.close();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+  }
 }

--- a/src/test/java/com/github/noconnor/junitperf/data/EvaluationContextTest.java
+++ b/src/test/java/com/github/noconnor/junitperf/data/EvaluationContextTest.java
@@ -12,6 +12,7 @@ import com.google.common.collect.ImmutableMap;
 
 import static java.util.Collections.emptyMap;
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static org.hamcrest.Matchers.anEmptyMap;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.startsWith;
 import static org.junit.Assert.assertThat;
@@ -200,6 +201,9 @@ public class EvaluationContextTest extends BaseTest {
     context.setStatistics(statisticsMock);
     context.runValidation();
     assertThat(context.isSuccessful(), is(true));
+    assertThat(context.isErrorThresholdAchieved(), is(true));
+    assertThat(context.isThroughputAchieved(), is(true));
+    assertThat(context.getPercentileResults(), is(anEmptyMap()));
   }
 
   @Test

